File: C++/5014_스타트링크.cpp
#include <iostream>
#include <queue>

using namespace std;

int N, s, e, U, D, visited[1000002];
queue<int> q;

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(0);
    cout.tie(0);

    cin >> N >> s >> e >> U >> D;

    q.push(s);
    visited[s] = 1;
    while (!q.empty()) {
        int cur = q.front();
        q.pop();
        if (cur == e) {
            cout << visited[cur] - 1;
            return 0;
        }
        for (int next : {cur + U, cur - D})
            if (0 < next && next <= N && !visited[next]) {
                q.push(next);
                visited[next] = visited[cur] + 1;
            }
    }
    cout << "use the stairs";
    return 0;
}

File: Python/5014_스타트링크.py
from collections import deque
input = __import__("sys").stdin.readline

N, start, end, U, D = map(int, input().split())
visited = [0] * 1000002
queue = deque([start])
visited[start] = 1

while queue:
    cur = queue.popleft()
    if cur == end:
        print(visited[end] - 1)
        break
    
    for nxt in [cur + U, cur - D]:
        if 0 < nxt <= N and not visited[nxt]:
            queue.append(nxt)
            visited[nxt] = visited[cur] + 1
else:
    print("use the stairs")
